token-metadata/program/src/utils.rs

```diff
@@ -213,6 +213,50 @@ pub fn create_or_allocate_account_raw<'a>(
     Ok(())
 }
 
+/// Resize an account using realloc, lifted from Solana Cookbook
+#[inline(always)]
+pub fn resize_or_reallocate_account_raw<'a>(
+    target_account: &AccountInfo<'a>,
+    funding_account: &AccountInfo<'a>,
+    system_program: &AccountInfo<'a>,
+    new_size: usize,
+) -> ProgramResult {
+    let rent = Rent::get()?;
+    let new_minimum_balance = rent.minimum_balance(new_size);
+
+    let lamports_diff = new_minimum_balance.saturating_sub(target_account.lamports());
+    invoke(
+        &system_instruction::transfer(funding_account.key, target_account.key, lamports_diff),
+        &[
+            funding_account.clone(),
+            target_account.clone(),
+            system_program.clone(),
+        ],
+    )?;
+
+    target_account.realloc(new_size, false)?;
+
+    Ok(())
+}
+
+/// Close src_account and transfer lamports to dst_account, lifted from Solana Cookbook
+#[inline(always)]
+pub fn close_account_raw<'a>(
```

Comment:
Nit: I wondered why this was called `raw` which made me think it didn't transfer out lamports.  Seems like it should just be called something like `close_account` or `empty_and_close_account`, but maybe you just want them named in accordance with other Solana cookbook functions?